lib/elixir/lib/macro.ex

```diff
@@ -624,7 +627,11 @@ defmodule Macro do
   # Splat when
   def to_string({:when, _, args} = ast, fun) do
     {left, right} = split_last(args)
-    fun.(ast, "(" <> Enum.map_join(left, ", ", &to_string(&1, fun)) <> ") when " <> to_string(right, fun))
+
+    fun.(
+      ast,
+      "(" <> Enum.map_join(left, ", ", &to_string(&1, fun)) <> ") when " <> to_string(right, fun)
```

Comment:
Can you please compute the result of the second argument on the previous line and call `fun.(ast, string)`? You will need to do similar changes in many clauses below. Thanks!